compiler/next/include/chpl/uast/ASTClassesList.h

```diff
@@ -95,12 +95,16 @@ AST_BEGIN_SUBCLASSES(Expression)       // old AST: Expr
 
   AST_END_SUBCLASSES(Loop)
 
-  //AST_NODE(ArrayExpr) // TODO: figure out good names for these
-  //AST_NODE(DomainExpr)
-  //AST_NODE(TupleExpr)
+  AST_NODE(Array)                      //
+  //AST_NODE(AssociativeArray)           //
+  AST_NODE(Domain)                     //
+  AST_NODE(Range)                      //
```

Comment:
Don't these also represent literals? Why not `ArrayLiteral` etc. and potentially make them extend `Literal`? Do we limit a `Literal` to a token always?

Reply:
I think you summarized it pretty well in your epiphany here: https://github.com/chapel-lang/chapel/pull/17975#discussion_r664917621

I think we want to reserve the term "literal" for things that can be statically evaluated and are more or less "immediates".

Reply:
Yeah, I was encouraging this terminology, based on what Wikipedia describes in the page about literals for programming languages.